src/oatpp/network/virtual_/client/ConnectionProvider.cpp

```diff
@@ -43,44 +45,68 @@ void ConnectionProvider::stop() {
 
 }
 
-std::shared_ptr<data::stream::IOStream> ConnectionProvider::get() {
-  auto submission = m_interface->connect();
+static std::shared_ptr<data::stream::IOStream> getSocket(std::shared_ptr<virtual_::Interface> interface,
+                                                         v_io_size maxAvailableToRead,
+                                                         v_io_size maxAvailableToWrite) {
+  auto submission = interface->connect();
   if(submission->isValid()) {
     auto socket = submission->getSocket();
     if (socket) {
       socket->setOutputStreamIOMode(oatpp::data::stream::IOMode::BLOCKING);
       socket->setInputStreamIOMode(oatpp::data::stream::IOMode::BLOCKING);
-      socket->setMaxAvailableToReadWrtie(m_maxAvailableToRead, m_maxAvailableToWrite);
+      socket->setMaxAvailableToReadWrtie(maxAvailableToRead, maxAvailableToWrite);
       return socket;
     }
   }
-  throw std::runtime_error("[oatpp::network::virtual_::client::getConnection()]: Error. Can't connect. " + m_interface->getName()->std_str());
+  throw std::runtime_error("[oatpp::network::virtual_::client::getConnection()]: Error. Can't connect. " + interface->getName()->std_str());
+}
+
+std::shared_ptr<data::stream::IOStream> ConnectionProvider::get(const std::chrono::duration<v_int64, std::micro>& timeout) {
+  std::packaged_task<std::shared_ptr<data::stream::IOStream>(std::shared_ptr<virtual_::Interface>, v_io_size, v_io_size)> task{getSocket};
+  auto future = task.get_future();
+
+  if (timeout == std::chrono::microseconds::zero()) {
+    task(m_interface, m_maxAvailableToRead, m_maxAvailableToWrite);
+    return future.get();
+  }
+
+  std::thread{std::move(task), m_interface, m_maxAvailableToRead, m_maxAvailableToWrite}.detach();
```

Comment:
We don't want to spawn threads in `ConnectionProvider::get` since it's too resource-inefficient.